src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/SwapGrTags.cpp

```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/SwapGrTags.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace grmhd::ValenciaDivClean::subcell {
+void SwapGrTags::apply(
+    const gsl::not_null<
+        Variables<typename System::spacetime_variables_tag::tags_list>*>
+        active_gr_vars,
+    const gsl::not_null<typename evolution::dg::subcell::Tags::Inactive<
+        typename System::spacetime_variables_tag>::type*>
+        inactive_gr_vars,
+    const Mesh<3>& dg_mesh,
+    const evolution::dg::subcell::ActiveGrid active_grid) noexcept {
+  if (active_grid == evolution::dg::subcell::ActiveGrid::Dg) {
+    if (active_gr_vars->number_of_grid_points() !=
+        dg_mesh.number_of_grid_points()) {
```

Comment:
Won't this function produce very unexpected results if `active_gr_vars-> number_of_grid_points()` is neither the number of dg grid points or subcell grid points? For instance, if I pass in an uninitialized vars...

Could there be an assertion that the vars are of the expected sizes before swapping?